### PR TITLE
Adding JS Challenge support

### PR DESCRIPTION
**Pull request by dimapx**
Also minor changes related to the configuration: 

- Removing default value from `sensitiveRoutes`.
- Setting the default for `sendPageActivities` to `true`.

### CODE REVIEW

File: PerimeterXModule/Internals/Helpers/PxConstants.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public static class PxConstants
 		public static readonly string LOG_CATEGORY = "PxModule";
 		public static readonly string MODULE_VERSION = GetAssemblyVersion();
 		public static readonly Options JSON_OPTIONS = new Options(prettyPrint: false, excludeNulls: true, includeInherited: true);
+        public static readonly string JS_CHALLENGE_ACTION = "j";
```
Comment (nitzanpx) [inline PerimeterXModule/Internals/Helpers/PxConstants.cs]: please fix indents :)

File: PerimeterXModule/Internals/PxContext.cs [hunk 2]
```diff
@@ -128,12 +128,17 @@ public PxContext(HttpContext context, PxModuleConfigurationSection pxConfigurati
 
 		private bool CheckSensitiveRoute(StringCollection sensitiveRoutes, string uri)
 		{
-			foreach( string sensitiveRoute in sensitiveRoutes) {
-				if (uri.StartsWith(sensitiveRoute))
+            if (sensitiveRoutes != null)
```
Comment (nitzanpx) [inline PerimeterXModule/Internals/PxContext.cs]: please fix indents

File: PerimeterXModule/Internals/Validators/PXS2SValidator.cs [hunk 3]
```diff
@@ -35,7 +35,13 @@ public bool VerifyS2S(PxContext PxContext)
 					PxContext.UUID = riskResponse.Uuid;
 					PxContext.BlockAction = riskResponse.RiskResponseAction;
 
-					if (score >= PxConfig.BlockingScore)
+                    if (PxContext.BlockAction == PxConstants.JS_CHALLENGE_ACTION &&
```
Comment (nitzanpx) [inline PerimeterXModule/Internals/Validators/PXS2SValidator.cs]: please fix indents :)

File: PerimeterXModule/PxModule.cs [hunk 4]
```diff
@@ -105,6 +106,7 @@ public PxModule()
 			appId = config.AppId;
 			suppressContentBlock = config.SuppressContentBlock;
 			captchaEnabled = config.CaptchaEnabled;
+            challengeEnabled = config.ChallengeEnabled;
```
Comment (nitzanpx) [inline PerimeterXModule/PxModule.cs]: please fix indents :)

File: PerimeterXModule/PxModule.cs [hunk 5]
```diff
@@ -292,12 +293,20 @@ private void ResponseBlockPage(PxContext pxContext)
 			var config = (PxModuleConfigurationSection)ConfigurationManager.GetSection(PxConstants.CONFIG_SECTION);
 			string template = "block";
 			string content;
-			if (captchaEnabled)
+            if (pxContext.BlockAction == "j")
```
Comment (nitzanpx) [inline PerimeterXModule/PxModule.cs]: please fix indents :)

File: PerimeterXModule/PxModule.cs [hunk 6]
```diff
@@ -292,12 +293,20 @@ private void ResponseBlockPage(PxContext pxContext)
...
+            {
+                template = "challenge";
+            }
+            else if (captchaEnabled)
```
Comment (nitzanpx) [inline PerimeterXModule/PxModule.cs]: new flow, captcha enabled should be deprecated
use BlockAction to decide what to render, j for challenge, c for captcha, else render block

Comment (dimapx) [reply PerimeterXModule/PxModule.cs]: Fixing as part of the [funCaptcha PR](https://github.com/PerimeterX/perimeterx-asp-net/pull/22): https://github.com/PerimeterX/perimeterx-asp-net/commit/c1e080a0909eabe7fed95367055e0376f0f2695f

**Pull request merged**